pkg/http/middleware/bodyfilter.go

```diff
@@ -0,0 +1,156 @@
+package middleware
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"path/filepath"
+
+	"golang.org/x/exp/slices"
+
+	"github.com/omissis/kube-apiserver-proxy/pkg/config"
+	http2 "github.com/omissis/kube-apiserver-proxy/pkg/http"
+)
+
+var ErrDuringBodyFilter = fmt.Errorf("error during body filter")
+
+func BodyFilterMux(conf []config.BodyFilterConfig) http2.MuxMiddleware {
+	return func(next http.Handler) http.Handler {
+		return BodyFilter(next, conf)
+	}
+}
+
+func BodyFilter(next http.Handler, conf []config.BodyFilterConfig) http2.Middleware {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		c, match := MatchConfig(r, conf)
+		if !match {
+			next.ServeHTTP(w, r)
+
+			return
+		}
+
+		if r.Body == nil {
+			http.Error(w, "Bad Request", http.StatusBadRequest)
+
+			return
+		}
+
+		bodyDecoder := json.NewDecoder(r.Body)
+
+		filteredBody := map[string]any{}
+		filterTarget := map[string]any{}
+
+		if err := bodyDecoder.Decode(&filteredBody); err != nil {
+			http.Error(w, "Bad Request", http.StatusBadRequest)
+
+			return
+		}
+
+		if err := json.Unmarshal([]byte(c.Filter), &filterTarget); err != nil {
+			http.Error(w, "Bad Request", http.StatusBadRequest)
+
+			return
+		}
+
+		if err := FillTargetFromBody(filteredBody, filterTarget); err != nil {
+			http.Error(w, "Bad Request", http.StatusBadRequest)
+
+			return
+		}
+
+		bodyFromTarget, err := json.Marshal(filterTarget)
+		if err != nil {
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+
+			return
+		}
+
+		r.Body = io.NopCloser(bytes.NewBuffer(bodyFromTarget))
+
+		next.ServeHTTP(w, r)
+	})
+}
+
+func MatchConfig(r *http.Request, conf []config.BodyFilterConfig) (config.BodyFilterConfig, bool) {
+	for _, c := range conf {
+		if !slices.Contains(c.Methods, r.Method) {
```

Comment:
it might be helpful to ensure the casing of both `c.Methods` and `r.Method` match, to avoid misses because one is uppercase and the other is not. I would ensure `c.Methods` are uppercase when the config gets loaded, and I would uppercase `r.Method` here.